interval.hh

```diff
@@ -615,8 +615,12 @@ public:
     }
     // Takes a vector of possibly overlapping intervals and returns a vector containing
     // a set of non-overlapping intervals covering the same values.
-    template<typename Comparator>
-    static std::vector<nonwrapping_interval> deoverlap(std::vector<nonwrapping_interval> intervals, Comparator&& cmp) {
+    template<typename Comparator, typename IntervalVec>
+    requires requires (IntervalVec vec) {
+        { vec.begin() } -> std::random_access_iterator;
+        { vec.end() } -> std::random_access_iterator;
```

Comment:
Add a constraint on IntervalVec::value_type, to ensure you get a vector of nonwrapping_interval and not a vector of int.

Reply:
You also need vec.reserve() to exist below, and maybe more.